packages/teams/test/utils/can-edit-team.test.ts

```diff
@@ -0,0 +1,67 @@
+import { canEditTeam } from "../../src/utils/can-edit-team";
+import { IGroup, IUser } from "@esri/arcgis-rest-types";
+
+describe("canEditTeam", function () {
+  const getUser = (props: any = {}) => props as IUser;
+  const getGroup = (props: any) => props as IGroup;
+
+  it("returns true if user is an admin or owner of the team", function () {
+    const groupId = "foo";
+    let group = getGroup({
+      id: groupId,
+      userMembership: {
+        memberType: "owner",
+        username: "scooby-doo",
+      },
+    });
+    const user = getUser({
+      groups: [group],
+      username: "scooby-doo",
+    });
+    let result = canEditTeam(group, user);
+    expect(result).toBe(true);
+    group = getGroup({
+      id: groupId,
+      userMembership: {
+        memberType: "admin",
+        username: "scooby-doo",
+      },
+    });
+    result = canEditTeam(group, user);
+    expect(result).toBe(true);
+  });
+
+  it("returns false if user is a member of the team", function () {
+    const groupId = "foo";
+    const group = getGroup({
+      id: groupId,
+      userMembership: {
+        memberType: "member",
+        username: "scooby-doo",
+      },
+    });
+    const user = getUser({
+      groups: [group],
+      username: "scooby-doo",
+    });
+    const result = canEditTeam(group, user);
+    expect(result).toBe(false);
+  });
+
+  it("returns false if user does not belong to the team", function () {
```

Comment:
This test is really checking if the group was fetched in the context of the user. If you just fetch a group w/o auth, there will be no `.userMembership` property returned b/c no user info was associated with the request. If you fetch a group as UserA, but then send that into the fn asking if UserB can edit the group... that's what we need to check/verify

I'd suggest renaming this test to `ensure group was fetched by the user` or something like that, and then add another test where you pass in a group what does not have `.userMembership` and ensure that always returns `false`. You'll also need to use `getProp` or `getWithDefault` to ensure the function does not error when you attempt to access child props of `.userMembership`